Problem creating under a package with UpperCases

Hi,

I found this error this morning using it. 

# Reproduce Step:
- Create a package named MyPackage
- Create a autovalue ```MyPackage.MyAutoValue.class```
```java
@AutoValue public abstract class MyAutoValue {
    @SerializedName("test") public abstract String test();
    @SerializedName("test_2") public abstract String test2();

    public static MyAutoValue create() {
        return null;
    }
}
```
-> this compiles

- add the TypeAdapter
```java
@AutoValue public abstract class MyAutoValue {
    @SerializedName("test") public abstract String test();
    @SerializedName("test_2") public abstract String test2();

    public static MyAutoValue create() {
        return null;
    }

    public static TypeAdapter<MyAutoValue> typeAdapter(Gson gson) {
        return new AutoValue_TestAutoValueGson.GsonTypeAdapter(gson);
    }
}
```

```
Error:(8, 40) error: cannot find symbol class MyPackage
Error:(18, 78) error: package MyPackage does not exist
Error:(26, 59) error: package MyPackage does not exist
Error:(35, 25) error: package MyPackage does not exist
```

I had another error looking like this
Could not find best guess class for <fullpath>.MyPackage.MyAutoValue.class on line 144

```java
   ClassName autoValueClass = ClassName.bestGuess(context.autoValueClass().getQualifiedName().toString());
```

# How to avoid the problem using this release
Put all packages in lowercase

# ?
Does get qualified name is returning a path with lowercase and another package is looking to an uppercase path?

It was an error by my part using an uppercase in a package but I had hard time to pinpoint the problem.

I hope it helps

--- a/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java
+++ b/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java
@@ -141,7 +141,7 @@ public class AutoValueGsonExtension extends AutoValueExtension {
     Map<String, TypeName> types = convertPropertiesToTypes(context.properties());
 
     ClassName classNameClass = ClassName.get(context.packageName(), className);
-    ClassName autoValueClass = ClassName.bestGuess(context.autoValueClass().getQualifiedName().toString());
+    ClassName autoValueClass = ClassName.get(context.autoValueClass());
 
     TypeSpec typeAdapter = createTypeAdapter(classNameClass, autoValueClass, properties);
 

--- a/src/test/java/com/ryanharter/auto/value/gson/AutoValueGsonExtensionTest.java
+++ b/src/test/java/com/ryanharter/auto/value/gson/AutoValueGsonExtensionTest.java
@@ -433,4 +433,26 @@ public class AutoValueGsonExtensionTest {
         .withWarningContaining("Found public static method returning TypeAdapter with no type "
             + "arguments, skipping GsonTypeAdapter generation.");
   }
+
+  @Test public void compilesWithCapitalPackageName() {
+    JavaFileObject source1 = JavaFileObjects.forSourceString("MyPackage.Foo", ""
+        + "package MyPackage;\n"
+        + "import com.google.auto.value.AutoValue;\n"
+        + "import com.google.gson.Gson;\n"
+        + "import com.google.gson.TypeAdapter;\n"
+        + "@AutoValue public abstract class Foo {\n"
+        + "  public static TypeAdapter<Foo> typeAdapter(Gson gson) {\n"
+        + "    return new AutoValue_Foo.GsonTypeAdapter(gson);"
+        + "  }\n"
+        + "  public abstract String a();\n"
+        + "  public abstract boolean b();\n"
+        + "}"
+    );
+
+    assertAbout(javaSource())
+        .that(source1)
+        .processedWith(new AutoValueProcessor())
+        .compilesWithoutError()
+        .withWarningCount(2);
+  }
 }

